src/components/UpdatePassword/UpdatePasswordButton.tsx

```diff
@@ -0,0 +1,129 @@
+import { useContext, useState } from "react";
+import { updatepasswordService } from "../../services/auth/updatepassword.service";
+import { FieldValues, useForm } from "react-hook-form";
+import { AuthContext } from "../../context/AuthContext";
+import toast from "react-hot-toast";
+
+export function UpdatePassword() {
+  const { session } = useContext(AuthContext);
+  const [showModal, setShowModal] = useState(false);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm();
+
+  interface UpdatepasswordRequest {
+    oldPassword: string;
+    newPassword: string;
+    token: string;
+  }
+
+  const onSubmit = async (formData: FieldValues, token: string) => {
+    const req: UpdatepasswordRequest = {
+      oldPassword: formData.password,
+      newPassword: formData.confirmPassword,
+      token: token
+    };
+    if (!errors.password && !errors.confirmPassword) {
+      try {
+        const response = await updatepasswordService(req);
+        if (response.success === true) {
+          toast.success("Password updated successfully");
+        } else {
+          toast.error("Failed to update password");
+        }
+      } catch (error) {
+        toast.error("Failed to update password");
+      }
+    }
+  };
+
+  return (
+    <>
+      <button
+        className="w-full rounded-lg border border-blue-600 bg-white p-2 text-center text-lg font-bold text-blue-600"
+        onClick={() => setShowModal(true)}
+      >
+        Update Password
+      </button>
+      {showModal ? (
+        <div className="absolute top-full mt-4 flex w-4/5 flex-col items-center justify-center gap-4 border bg-white p-4">
+          <form
+            onSubmit={handleSubmit((formData) =>
+              onSubmit(formData, session!.token)
+            )}
+            className="g-2 w-full"
+          >
+            <p className="m-2 ml-2 text-center text-2xl font-bold text-blue-600">
+              Update Password
+            </p>
+            <input
```

Comment:
As the inputs does not have a visible `<label />`, consider using an `aria-label` attribute.